### PR TITLE
Summer 2021 Blogpost

### PR DESCRIPTION
**Pull request by sandertyu**
Here is the initial draft for our blogpost. I haven't done a very intense spell/grammer check, and if Tim has any further things to add, they may do so before merging.

### CODE REVIEW

File: content/jupyter-summer-2021.rst [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+:summary: Blog post on the progress made by JupyterTeam since winter 2020 through summer 2021
+
+Improvements and Progress
+-------------------------
+
+Since winter quarter of 2020, the JupyterTeam has fixed a long standing issue where Thebe,
+the Javascript package that we use to display and run code on libretexts.org, would not properly
+work with the entire interactive Jupyter widget ecosystem. By migrating from the 
+Jupyter HTML manager package to the Lab manager, we are now able to run all 
+ipywidget interactive widgets, which greatly expands the possibilities for textbook authors.
```
Comment (moorepants) [inline content/jupyter-summer-2021.rst:19]: I'd love to see one of the gifs Tim made (or a new one) showing this working in libretexts.

File: content/jupyter-summer-2021.rst [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+available with the already existing one; if the master router fails for any reason,
+then the backup one will take its place until the master comes back online. Furthermore,
+we configured the routers to act as TFPT servers so that they we may use Ubuntu netboot
+to load Ubuntu on all nodes from scratch. As compared to the previous method, this
+way is more effective and utilizes existing infastructure better.
+
+We completed the migration of our default-env computing environment
+to JupyterLab 3.0, which includes new features which are great for widgets
+and their Javascript extensions. Additionally, we added a new Adapt plugin
+to supplement our already existing Query plugin on libretexts.org.
```
Comment (moorepants) [inline content/jupyter-summer-2021.rst:31]: Any images that could be added for these two things that show off the new aspects?

File: content/jupyter-summer-2021.rst [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+Finally, after many bitcoining mining attempts on libretexts.org through our binder.libretexts.org,
+we have implemented successful automated measures in order to deal with these cryptominers
+and prevent them from mining on the cluster. Since then, there have been no major instances
+of bitcoin miners running on the cluster, and we have further ideas if more
+anti-mining measures need to be added in the future.
```
Comment (moorepants) [inline content/jupyter-summer-2021.rst:37]: I think we should leave this off. The more clues you give to the miners, the more they might take advantage. Maybe a minimal 1 sentence that says "We improved abuse mitigations on our binder instance" or something like that is plenty.

File: content/jupyter-summer-2021.rst (hunk 1 above)
Comment (moorepants) [reply content/jupyter-summer-2021.rst:19]: @timjstewart

Comment (moorepants) [reply content/jupyter-summer-2021.rst:19]: @TimStewartJ

Comment (TimStewartJ) [reply content/jupyter-summer-2021.rst:19]: I've added a gif for this, let me know what you think of it.

Comment (moorepants) [reply content/jupyter-summer-2021.rst:19]: Looks good! Thanks,

File: content/jupyter-summer-2021.rst [hunk 4]
```diff
@@ -33,11 +33,8 @@ to JupyterLab 3.0, which includes new features which are great for widgets
 and their JavaScript extensions. Additionally, we added a new Adapt plugin
 to supplement our already existing Query plugin on libretexts.org.
 
-Finally, after many bitcoining mining attempts on libretexts.org through our binder.libretexts.org,
-we have implemented successful automated measures in order to deal with these cryptominers
-and prevent them from mining on the cluster. Since then, there have been no major instances
-of bitcoin miners running on the cluster, and we have further ideas if more
-anti-mining measures need to be added in the future.
+Finally, after many bitcoining mining attempts on libretexts.org,
```
Comment (moorepants) [inline content/jupyter-summer-2021.rst:36]: Sorry, but I don't think we should say bitcoin or mining at all. Let's just remove this. Or make it very vague as I suggested.

File: content/jupyter-summer-2021.rst [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+-------------------------
+
+Since winter quarter of 2020, the JupyterTeam has fixed a long standing issue where Thebe,
+the JavaScript package that we use to display and run code on libretexts.org, would not properly
+work with the entire interactive Jupyter widget ecosystem. By migrating from the 
+Jupyter HTML Manager package to the Lab Manager, we are now able to run all 
+ipywidget interactive widgets, which greatly expands the possibilities for textbook authors.
+
+.. |thebe-working| image:: https://cdn.discordapp.com/attachments/840776718930411561/885315459560374323/BdXwA3V0Pl.gif
+   :height: 200px
```
Comment (moorepants) [inline content/jupyter-summer-2021.rst:22]:
```suggestion
   :height: 200px
   :align: center
```

File: content/jupyter-summer-2021.rst [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+Finally, after many bitcoining mining attempts on libretexts.org,
+we have successfully implemented measures to mitigate these abuses.
+
+In the Works
+------------
+
+For the future, we are planning to develop a Ceph cluster to replace our NFS-client-provisioner 
+as the storage solution for users on the JupyterHub. We also intend to extend the cluster
+with GPU nodes and provide more specialized computing enviornments. 
+Feel free to use our Binder plugin on `libretexts.org <https://libretexts.org/>`__, and
```
Comment (moorepants) [inline content/jupyter-summer-2021.rst:45]:
```suggestion
Feel free to use our Binder plugin on `libretexts.org <https://libretexts.org/>`_, and 
```

File: content/jupyter-summer-2021.rst [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+we have successfully implemented measures to mitigate these abuses.
+
+In the Works
+------------
+
+For the future, we are planning to develop a Ceph cluster to replace our NFS-client-provisioner 
+as the storage solution for users on the JupyterHub. We also intend to extend the cluster
+with GPU nodes and provide more specialized computing enviornments. 
+Feel free to use our Binder plugin on `libretexts.org <https://libretexts.org/>`__, and 
+check out our `JupyterHub <https://jupyterhub.ucdavis.edu>`__ if you are a UC Davis member!
```
Comment (moorepants) [inline content/jupyter-summer-2021.rst:46]:
```suggestion
check out our `JupyterHub <https://jupyterhub.ucdavis.edu>`_ if you are a UC Davis member!
```

File: content/jupyter-summer-2021.rst [hunk 8]
```diff
@@ -0,0 +1,46 @@
+==============================
```
Comment (moorepants) [inline content/jupyter-summer-2021.rst:1]:
```suggestion
================================
```

File: content/jupyter-summer-2021.rst [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+JupyterTeam Progress Summer 2021
+==============================
```
Comment (moorepants) [inline content/jupyter-summer-2021.rst:3]:
```suggestion
================================
```

File: content/jupyter-summer-2021.rst [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+Improvements and Progress
+-------------------------
+
+Since winter quarter of 2020, the JupyterTeam has fixed a long standing issue where Thebe,
+the JavaScript package that we use to display and run code on libretexts.org, would not properly
+work with the entire interactive Jupyter widget ecosystem. By migrating from the 
+Jupyter HTML Manager package to the Lab Manager, we are now able to run all 
+ipywidget interactive widgets, which greatly expands the possibilities for textbook authors.
+
+.. |thebe-working| image:: https://cdn.discordapp.com/attachments/840776718930411561/885315459560374323/BdXwA3V0Pl.gif
```
Comment (moorepants) [inline content/jupyter-summer-2021.rst:21]:
```suggestion
.. image:: https://objects-us-east-1.dream.io/mechmotum/thebe-ipywidgets.gif
```

File: content/jupyter-summer-2021.rst [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+-------------------------
+
+Since winter quarter of 2020, the JupyterTeam has fixed a long standing issue where Thebe,
+the JavaScript package that we use to display and run code on libretexts.org, would not properly
+work with the entire interactive Jupyter widget ecosystem. By migrating from the 
+Jupyter HTML Manager package to the Lab Manager, we are now able to run all 
+ipywidget interactive widgets, which greatly expands the possibilities for textbook authors.
+
+.. image:: https://objects-us-east-1.dream.io/mechmotum/thebe-ipywidgets.gif
+   :height: 200px
```
Comment (moorepants) [inline content/jupyter-summer-2021.rst:22]:
```suggestion
   :height: 400px
```

File: content/jupyter-summer-2021.rst [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+Improvements and Progress
+-------------------------
+
+Since winter quarter of 2020, the JupyterTeam has fixed a long standing issue where Thebe,
+the JavaScript package that we use to display and run code on libretexts.org, would not properly
+work with the entire interactive Jupyter widget ecosystem. By migrating from the 
+Jupyter HTML Manager package to the Lab Manager, we are now able to run all 
+ipywidget interactive widgets, which greatly expands the possibilities for textbook authors.
+
+.. image:: https://objects-us-east-1.dream.io/mechmotum/thebe-ipywidgets.gif
```
Comment (moorepants) [inline content/jupyter-summer-2021.rst:21]:
```suggestion
.. figure:: https://objects-us-east-1.dream.io/mechmotum/thebe-ipywidgets.gif
```

File: content/jupyter-summer-2021.rst [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+   :height: 400px
+   :align: center
+
```
Comment (moorepants) [inline content/jupyter-summer-2021.rst:24]:
```suggestion

   Example of an ipywidget controlling a matplotlib figure in Thebe.
```

File: content/jupyter-summer-2021.rst [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+work with the entire interactive Jupyter widget ecosystem. By migrating from the 
+Jupyter HTML Manager package to the Lab Manager, we are now able to run all 
+ipywidget interactive widgets, which greatly expands the possibilities for textbook authors.
+
+.. figure:: https://objects-us-east-1.dream.io/mechmotum/thebe-ipywidgets.gif
+   :height: 400px
+   :align: center
+
+   Example of an ipywidget controlling a matplotlib figure in Thebe.
+In terms of cluster reliability, we have added a second router to be highly
```
Comment (moorepants) [inline content/jupyter-summer-2021.rst:26]:
```suggestion

In terms of cluster reliability, we have added a second router to be highly 
```

**Pull request merged**